### PR TITLE
Pokebilities: fix special abilities as well as Gastro Acid

### PR DESCRIPTION
**Pull request by MacChaeger**
@urkerab does this look right?

### CODE REVIEW

File: data/pokedex.js [hunk 1]
```diff
@@ -11472,7 +11472,7 @@ let BattlePokedex = {
 		types: ["Dragon", "Ground"],
 		gender: "N",
 		baseStats: {hp: 108, atk: 100, def: 121, spa: 81, spd: 95, spe: 95},
-		abilities: {0: "Aura Break", H: "Power Construct"},
+		abilities: {0: "Aura Break", S: "Power Construct"},
```
Comment (Zarel) [inline data/pokedex.js]: No, Power Construct is definitely Zygarde's Hidden Ability.

Comment (Zarel) [reply data/pokedex.js]: `S` means "alternate forme's primary ability".

Comment (MacChaeger) [reply data/pokedex.js]: Check the mechanics discord, #showdown-coding

File: config/formats.js [hunk 2]
```diff
@@ -574,7 +574,7 @@ let Formats = [
 		onBegin: function () {
 			let allPokemon = this.p1.pokemon.concat(this.p2.pokemon);
 			for (let pokemon of allPokemon) {
-				if (pokemon.ability === pokemon.template.abilities['S']) {
+				if (pokemon.ability === toId(pokemon.template.abilities['S'])) {
```
Comment (urkerab) [inline config/formats.js]: Ugh, good catch!

File: data/mods/pokebilities/moves.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+/**@type {{[k: string]: ModdedMoveData}} */
+exports.BattleMovedex = {
+	"gastroacid": {
+		inherit: true,
+		effect: {
+			// Ability suppression implemented in Pokemon.ignoringAbility() within sim/pokemon.js
+			onStart: function (pokemon) {
+				this.add('-endability', pokemon);
+				this.singleEvent('End', this.getAbility(pokemon.ability), pokemon.abilityData, pokemon, pokemon, 'gastroacid');
+				if (pokemon.innates) pokemon.innates.forEach(innate => pokemon.removeVolatile("ability" + innate));
```
Comment (urkerab) [inline data/mods/pokebilities/moves.js]: Oops, I forgot to remove `forEach` from `onAfterMega`...

File: data/pokedex.js (hunk 1 above)
Comment (Zarel) [reply data/pokedex.js]: My mistake, I forgot that Zygarde-50 was Zygarde.

**Comment by MacChaeger** [comment]
What is this waiting on?

**Pull request merged**